### ISSUE TITLE
Verify that the user gets assigned on PR open

### ISSUE THREAD

**Issue by gentlementlegen** [ubiquibot/command-start-stop#55]
@Keyrxng rfc

What I think happened is:
- no user was assigned
- a pull-request was linked to this issue, which triggers an assignment of the user opening that PR
- the user somehow didn't get assigned (you can see from the timeline only the start message was triggered)
- `/start` assigned the user and displayed the message again

It should have just said "You are already assigned to this task" because it should have been the case.

_Originally posted by @gentlementlegen in https://github.com/ubiquibot/assistive-pricing/issues/32#issuecomment-2362687081_

**Comment by 0x4007** [comment]
I think that the logic of v1 needs to be replicated more carefully. 

We keep having to make iterative fixes on this plugin even though it was already implemented correctly in v1. 

I don't understand why we are having these problems.

**Comment by gentlementlegen** [comment]
@0x4007 More of a reminder than anything, so we don't forget. I'll take care of it.

**Comment by gentlementlegen** [comment]
Updates on this ticket:

I tested every scenario and could not reproduce the issue. Even if an assignment fails, an error message will be posting saying it failed (and no start message) which definitely means there was no error during the workflow run. I tested:
- `/start` command
- linking a pull-request ad a draft / as a ready to review
- with outside collaborators

They all worked fine. That's when I noticed on the issue where we had an error that the timeline for the events all happen within the same minute. So I looked in detail and there was less than 20 seconds between each event. The only problem I noticed and when you open a PR, then change the description and link an issue, the bot won't assign and post the comment, which believe should be fixed (dunno if v1 handled that).

So my second try, I linked a PR and did a `/start` command with less than 3 seconds interval, and this is the result:
https://github.com/Meniole/command-start-stop/issues/32

We can see it's extremely similar to what we saw on our organization. What happens is that both workflows run very closely and GitHub didn't update the timeline yet, so both will post the start message but only one assignment will succeed. I can investigate more to see if there is any workaround on this, but feels a bit like an edge case.
